dali/python/nvidia/dali/pipeline.py

```diff
@@ -555,21 +558,29 @@ def to_numpy(x):
         # __cuda_array_interface__ doesn't provide any way to pass the information about the device
         # where the memory is located. It is assumed that the current device is the one that the memory belongs to,
         # unless the user sets the device explicitly creating TensorGPU/TensorListGPU
-        if isinstance(data, list):
+        if isinstance(data, (Tensors.TensorListCPU, Tensors.TensorListGPU)):
+            if layout is not None:
```

Comment:
All looks correct, but I can't shake the feel that I just observed at least 3 distinct ways for handling the `None` layout. And, doesn't the TL constructor at least also handle the `None` layout only overwriting it if it was actually provided?

I'm not sure if I can suggest any better way of handling the `None` layout vs the default empty one. Maybe it should still be handled by the TL/T constructor.